app/src/main/java/com/example/weatherapp/ForecastActivity.kt

```diff
@@ -0,0 +1,118 @@
+package com.example.weatherapp
+
+import android.annotation.SuppressLint
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.w3c.dom.Text
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.format.DateTimeFormatter
+
+
+ class ForecastActivity : AppCompatActivity() {
+
+    private lateinit var recyclerView: RecyclerView
+
+    private val adapterData = mutableListOf<DayForecast>()
+
+
+
+    private val apiKey = "43d3f89c03787217902a3910dc8baaa0"
+    private lateinit var api: weatherApi
+
+    private lateinit var date: TextView
+    private lateinit var sunrise: TextView
+    private lateinit var sunset: TextView
+    private lateinit var day: TextView
+    private lateinit var minTemp: TextView
+    private lateinit var maxTemp: TextView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_forecast)
+
+        date = findViewById(R.id.date)
```

Comment:
Your app crashes because there is no view with the id `date` in your activity_forecast.xml. Remove lines 34-39, 46-50, and 95-112.